orttraining/orttraining/python/training/ortmodule/_torch_module_factory.py

```diff
@@ -2,13 +2,18 @@
 # Licensed under the MIT License.
 # _torch_module_factory.py
 
+from ._io import _FlattenedModule
 from ._torch_module_ort import TorchModuleORT
-from .debug_options import DebugOptions
-from ._fallback import _FallbackManager
+from ._graph_execution_manager import GraphExecutionManager
+
+import torch
 
 
 class TorchModuleFactory:
-    def __call__(self, module, debug_options: DebugOptions, fallback_manager: _FallbackManager):
+    def __call__(self,
+                 module: torch.nn.Module,
+                 flattened_module: _FlattenedModule,
```

Comment:
Is it possible to avoid passing in the `_FlattenedModule` here? The idea of the `TorchModuleFactory` was to create a any type of `TorchModule` as defined by the input module. For example, if the input module is a `torch.nn.Module`, then we must instantiate the `TorchModuleORT` class. If the input module is an onnx model, we must instantiate the `TorchOnnxModule` and so on.
The function should only take in the module and return the torch module for that object.
Not all torch modules would need the `flattened_module`, since it is very specific to the torch ort module.
So, can we keep the specialization in the graph managers or in the torch modules and not leak that implementation here?

Reply:
My idea was to keep ORTModule aware of the flattenes, but you are right. Not all variants would use flattened module